src/client/components/layouts/Header/Navigation/NavItemWithSubItems.test.tsx

```diff
@@ -0,0 +1,200 @@
+/**
+ @jest-environment jsdom
+ */
+
+import '@testing-library/jest-dom';
+
+import { fireEvent,render, screen } from '@testing-library/react';
+import { userEvent } from '@testing-library/user-event';
+import React from 'react';
+
+import { KeyBoard } from '~/client/components/keyboard/keyboard.enum';
+import { mockUseRouter } from '~/client/components/useRouter.mock';
+
+import { NavItemWithSubItems } from './NavItemWithSubItems';
+
+const mockNavigationItemWithChildren = {
+	children: [
+		{ label: 'Sub Item 1', link: '/sub1' },
+		{ label: 'Sub Item 2', link: '/sub2' },
+	],
+	label: 'Test Menu',
+};
+
+describe('NavItemWithSubItems', () => {
+	beforeAll(() => {
+		Object.defineProperty(window, 'location', {
+			value: { 
+				assign: jest.fn(), 
+			},
+		});
+		mockUseRouter({ pathname: '/' });
+	});
+
+	it('doit rendre le bouton du menu', () => {
+		render(<NavItemWithSubItems navigationItemWithChildren={mockNavigationItemWithChildren} />);
+		
+		expect(screen.getByRole('button', { name: 'Test Menu' })).toBeVisible();
+	});
+
+	describe('lorsque je clique sur le bouton du menu', () => {
+		it('doit afficher l‘attribut aria-expanded correctement', async () => {
+			const user = userEvent.setup();
+
+			render(<NavItemWithSubItems navigationItemWithChildren={mockNavigationItemWithChildren} />);
+			const menuButton = screen.getByRole('button', { name: 'Test Menu' });
+      
+			expect(menuButton).toHaveAttribute('aria-expanded', 'false');
+      
+			await user.click(menuButton);
+			expect(menuButton).toHaveAttribute('aria-expanded', 'true');
+		});
+
+		it('doit jongler sur la visibilité du menu au clic', async () => {
+			const user = userEvent.setup();
+
+			render(<NavItemWithSubItems navigationItemWithChildren={mockNavigationItemWithChildren} />);
+			const menuButton = screen.getByRole('button', { name: 'Test Menu' });
+      
+			await user.click(menuButton);
+			expect(screen.getByRole('link', { name: /Sub Item 1/i })).toBeVisible();
+      
+			await user.click(menuButton);
+
+			expect(screen.queryByText('Sub Item 1')).not.toBeInTheDocument();
+		});
+
+		describe('lorsqu‘un élément du sous-menu est cliqué', () => {
```

Comment:
On pourrait se passer du describe ici, vu qu'on a qu'un seul it 

it('lorsqu‘un élément du sous-menu est cliqué, appelle la props onClick une fois' ...